qiskit_experiments/base_analysis.py

```diff
@@ -87,33 +89,54 @@ def run(
         analysis_options = analysis_options.__dict__
 
         # Run analysis
+        analysis_result_parameters = {
+            "result_type": experiment_data.experiment_type,
+            "experiment_id": experiment_data.experiment_id,
+        }
+        if "physical_qubits" in experiment_data.metadata():
+            analysis_result_parameters["device_components"] = [
```

Comment:
For some experiments the device components are not the qubits (see #171). So what's written here is probably temporary, and after #171 probably each experiment will populate the device components in its result data (in _run_analysis) and we will fetch them from there.